sourcemash/admin.py

```diff
@@ -0,0 +1,27 @@
+from flask import redirect, url_for
+from flask.ext.admin import Admin
+from flask.ext.admin.contrib.sqla import ModelView
+from flask.ext.security import current_user
+from sourcemash.database import db
+from sourcemash.models import Feed
+
+admin = Admin()
+
+
+class FeedView(ModelView):
+    can_create = False
+    can_delete = False
+
+    def is_accessible(self):
+        return current_user.is_authenticated() and \
+               current_user.email == "admin@sourcemash.com"
+
+    def _handle_view(self, name, **kwargs):
+            if not self.is_accessible():
+                return redirect(url_for('security.login'))
+
+    column_display_pk = True
+    form_columns = ['title', 'topic', 'public', 'image_url']
+
+
+admin.add_view(FeedView(Feed, db.session))
```

Comment:
Yessir

Alex Gerstein
gerstein.alex@gmail.com




On Wed, May 13, 2015 at 9:56 AM -0700, "Scott Gladstone" <notifications@github.com> wrote:












In sourcemash/admin.py:
> +    can_create = False
> +    can_delete = False
> +
> +    def is_accessible(self):
> +        return current_user.is_authenticated() and \
> +               current_user.email == "admin@sourcemash.com"
> +
> +    def _handle_view(self, name, **kwargs):
> +            if not self.is_accessible():
> +                return redirect(url_for('security.login'))
> +
> +    column_display_pk = True
> +    form_columns = ['title', 'topic', 'public', 'image_url']
> +
> +
> +admin.add_view(FeedView(Feed, db.session))



Is the admin page at /admin? 



—
Reply to this email directly or view it on GitHub.